content/git/concepts/gitignore/gitignore.md

```diff
@@ -0,0 +1,96 @@
+```
+src/main.js
+
+```
+to ignore the file **main.js** under the **src/** directory.
+
+![alt text](https://static-assets.codecademy.com/Courses/learn-git-github/gitignore/create-gitignore.png "A screenshot of a Linux terminal showing the creation of a .gitignore file using the command "touch .gitignore". Next, the commands "ls" and "ls -a" are used to show that since .gitignore is a hidden file, the "-a" flag is necessary")
+
+**Note:** since **.gitignore** is a hidden file, we will need to add the `-a` flag to `ls` to see it.
+
+## Ignore a directory with .gitignore
+
+Sometimes we want to ignore entire directories or specify certain files in a directory. Common directories to leave out of a Git repository are **node_modules** or **logs** folder. We can ignore an entire directory by simply adding its name to **.gitignore**:
+```
+node_modules/
+```
+This will ignore the **node_modules** directory, and all subdirectories and files inside them. The forward slash `/` specifies that we are ignoring the directory.
+
+## `.gitignore` Patterns
+
+We can take advantage of patterns to match multiple filenames. These help us handle special cases such as ignoring specific file types or ignoring all but one file inside a directory. Some examples of things that make up patterns are:
+
+* Wildcard * to match 0 or more characters except for /. For example, adding `*.html` to **.gitignore** would ignore all files ending with the `.html` extension. `example*` would match any file starting with example such as `example.txt` or `exampleHtmlFile.html`.
+* Negation `!` as a prefix to negate any file that would otherwise be ignored. For example,
+```
+index*
+!public/index.css
+```
+will ignore all files starting with *index* except for *src/index.css*. But, we cannot negate a file inside an ignored directory.
+
+* Square brackets `[]` can be used to match a single character from a set of characters or a range of characters. Note that the range can be alphabetical: `[a-z]` or `[A-Z]`, numeric `[0-9]`, or a set of characters. If we added `index.[a-i]*` with both the square bracket and wildcard to **.gitignore**, we would ignore *index.css* and *index.html* but not *index.js*, since “j” is outside of the `[a-i]` range.
+* Double asterisk `**` is used to match 0 or more directories. If we had a temp folder inside all of the folders in the root directory and we only wanted to match files with the `.log` extension, we could use the pattern `**/temp/*.log`.
+
+## GitHub Provided Templates
+
+When we create a new repository on GitHub, we have the option to add a .gitignore file from a list of templates. These templates are pulled from [GitHub’s gitignore repository](https://github.com/github/gitignore). For example, below is the template for Java projects.
+```
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
+```
```

Comment:
This example can be skipped as we have already added the link where learners can find the template.

Reply:
```suggestion
```